opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/__init__.py

```diff
@@ -19,7 +19,7 @@
 from re import sub
 from shutil import which
 
-from pkg_resources import iter_entry_points
```

Comment:
you might want to remove the `setuptools` dependency in the [pyproject.toml](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/opentelemetry-instrumentation/pyproject.toml#L28) file since it shouldn't be required anymore.